packages/components/tabs/src/useResizeObserver.ts

```diff
@@ -0,0 +1,42 @@
+import { type RefObject, useEffect, useRef, useState } from 'react'
+
+interface Size {
+  width?: number
+  height?: number
+}
+
+type ResizeCallback = (entry?: ResizeObserverEntry) => void
+
+export const useResizeObserver = <T extends HTMLElement>(
+  target: RefObject<T> | T | null,
+  onResize?: ResizeCallback
+): Size => {
+  const [size, setSize] = useState<Size>({ width: undefined, height: undefined })
+  const resizeObserverRef = useRef<ResizeObserver>()
+  const resizeCallbackRef = useRef<ResizeCallback | undefined>(onResize)
+
```

Comment:
see -> https://tkdodo.eu/blog/refs-events-and-escape-hatches
```suggestion
// 👇 update the ref when the callback changes
  React.useEffect(() => {
    resizeCallbackRef.current = onResize
  }, [onResize])
```

Reply:
If I'm not wrong, there is no need here, as the callback is a prop. So whenever it changes, the ref also will?

Reply:
Nope, see -> https://epicreact.dev/why-you-shouldnt-put-refs-in-a-dependency-array/

> with code like this:
> ```tsx
> function Comp() {
>   const logRef = React.useRef(console.log)
>   React.useEffect(() => {
>     logRef.current(new Date())
>   }, [logRef]) // <-- ✅ No eslint warning here
>   return <div>{/* stuff here */}</div>
> }
> ```
> That's because the logRef value can never change. **So it actually makes no difference whether you include it or not** and I guess the authors of the lint rule decided to not bother you about something that doesn't matter. Good call.

To get a better idea of why you need to update your ref, see this [codesandbox](https://codesandbox.io/s/late-wind-ht22gy?file=/src/pages/Playground/index.tsx) I've made that illustrates the problem:
1. increment the counter to to a value greater than 10 by clicking on the button
2. check the logs (you'll notice there is a problem) ❌
3. now, uncomment lines 13-15
4. refresh the view
5. increment the counter to a value greater than 10 by clicking on the button
6. check the logs (things will be working as intended) ✅